models/SymbolMaster/SymbolMaster.test.js

```diff
@@ -63,4 +67,239 @@ describe('SymbolMaster', () => {
+        },
+      ],
+    });
+
+    expect(symbolInstance.overrideValues[0].value).toEqual(overrideText);
+  });
+
+  it('should be able to create a symbol instance with layer style overrides', () => {
+    const beforeStyle = new SharedStyle({
+      name: 'before',
+      fills: [
+        {
+          color: '#fff',
+        },
+      ],
+    });
+
+    const afterStyle = new SharedStyle({
+      name: 'after',
+      fills: [
+        {
+          color: '#000',
+        },
+      ],
+    });
+
+    const symbolMaster = new SymbolMaster({
+      name: 'symbol',
+    });
+
+    const layer = new Rectangle({
+      width: 200,
+      height: 100,
+      x: 0,
+      y: 0,
+      name: 'rectangle',
+      style: beforeStyle,
+    });
+
+    symbolMaster.addLayer(layer, true);
+
+    // symbolMaster.addLayer currenty sets the overrideName incorrectly - temp fix
+    symbolMaster.overrideProperties[0].overrideName = symbolMaster.overrideProperties[0].overrideName.replace(
+      'stringValue',
+      'layerStyle'
+    );
+
+    const symbolInstance = symbolMaster.createInstance({
+      overrideValues: [
+        {
+          name: 'rectangle',
+          value: afterStyle,
+        },
+      ],
+    });
+
+    expect(symbolInstance.overrideValues[0].overrideName).toEqual(symbolMaster.overrideProperties[0].overrideName);
+    expect(symbolInstance.overrideValues[0].value).toEqual(afterStyle.do_objectID);
+  });
+
+  it('should be able to create a symbol instance with overrides within groups', () => {
+    const symbolMaster = new SymbolMaster({
+      name: 'symbol',
+    });
+
+    const group = new Group({
+      name: 'group',
+    });
+
+    const text = new Text({
+      name: 'text',
+    });
+
+    const overrideText = 'override text';
+
+    group.addLayer(text);
+
+    symbolMaster.addLayer(group, true);
+
+    // symbolMaster.addLayer currenty sets the overrideProperties incorrectly - temp fix
+    symbolMaster.overrideProperties[0].overrideName = `${text.do_objectID}_stringValue`;
+
+    const symbolInstance = symbolMaster.createInstance({
+      overrideValues: [
+        {
+          name: 'text',
+          value: overrideText,
+        },
+      ],
+    });
+
+    expect(symbolInstance.overrideValues[0].overrideName).toEqual(symbolMaster.overrideProperties[0].overrideName);
+    expect(symbolInstance.overrideValues[0].value).toEqual(overrideText);
+  });
+
+  it('should be able to update simple text Symbol Instance override', () => {
+    const symbolMaster = new SymbolMaster({
+      name: 'symbol',
+    });
+
+    const layer = new Text({
+      name: 'text',
+      string: 'before',
+    });
+
+    const overrideText = 'after';
+
+    symbolMaster.addLayer(layer, true);
+    const symbolInstance = symbolMaster.createInstance({ name: 'instance' });
+
+    symbolMaster.updateInstance(symbolInstance, [{ name: 'text', value: overrideText }]);
+
+    console.log(`result: ${symbolInstance.overrideValues[0].value}`);
```

Comment:
Could we remove this if it isn't necessary for testing?

Reply:
Yup. That was a mistake. Didn't mean to include it.

Reply:
removed